### DISCUSSION TITLE
Why janusgraph publish so many minor version instead of snapshot verson?

### DISCUSSION THREAD

**Discussion by sonnyhcl** [JanusGraph/janusgraph#4001] [General]
As below shows, Why janusgraph publish so many minor version instead of snapshot verson? I have a project depending on janusgraph and receive dependabot PR update on janusgraph so frequent! Isn't it kind of abuse?

![image](https://github.com/JanusGraph/janusgraph/assets/10840522/6fe44ebf-dbe1-4df5-808a-d5fa85d31cf5)

**Comment by li-boxuan** [comment]
To be fair, isn't dependabot rather than JanusGraph that's "abusing"? :)

You may want to try this: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates#disabling-dependabot-version-updates

As to your original question: JanusGraph publishes a minor version per commit because JanusGraph community does not have enough bandwidth to do (relatively) frequent official releases (we do it roughly once a year). To make users able to use latest features & bug fixes, a CI/CD process is configured so that every commit is published on maven.

**Comment by sonnyhcl** [comment]
Thanks for the context.
